engines/crab/ui/FileData.cpp

```diff
@@ -0,0 +1,93 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*
+ * This code is based on the CRAB engine
+ *
+ * Copyright (c) Arvind Raja Yadav
+ *
+ * Licensed under MIT
+ *
+ */
+
+#include "common/savefile.h"
+#include "crab/crab.h"
+#include "crab/loaders.h"
+#include "crab/XMLDoc.h"
+#include "crab/ui/FileData.h"
+
+namespace Crab {
+
+using namespace pyrodactyl::ui;
+
+SaveFileData::SaveFileData(const Common::String &file) {
+	_path = file;
+
+	// Extract String between _ and . For eg., CRAB_Autosave 1.unr -> Autosave 1
+	// 4 => .unr
+	size_t pos = file.findFirstOf('_');
+	_name = file.substr(pos + 1, file.size() - (pos + 1) - 4);
```

Comment:
What are the savefile names? Normally, it should be `target.s###` in ScummVM, so you know which target (not gameid) it is. And autosave is a separate subsystem in scummvm, those go to slot #0.

Reply:
Savefile names are "Unrest.###" for the full game and  "Unrest-demo.###" for the demo.